libs/features/personalization/personalization.js

```diff
@@ -424,24 +403,39 @@ const addHash = (url, newHash) => {
   }
 };
 
+const setDataIdOnChildren = (sections, id, value) => {
+  [...sections[0].children].forEach(
```

Comment:
do we know that sections[0] will always exist ?

Reply:
@denlight, @vivgoodrich, setDataIdOnChildren is only called within the updateFragDataProps function which is only called in the fragment.js init function. There is already a check for sections.length before the sections variable is passed into updateFragDataProps (fragment.js, line 102) so the current code is safe. I can come back to this in a future ticket, but I suggest we leave it as is tonight to avoid last second changes. Thanks.